client/me/concierge/info-step.js

```diff
@@ -4,38 +4,129 @@
  * External dependencies
  */
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import moment from 'moment-timezone';
 
 /**
  * Internal dependencies
  */
-import Button from 'components/button';
 import CompactCard from 'components/card/compact';
+import FormButton from 'components/forms/form-button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextarea from 'components/forms/form-textarea';
+import Timezone from 'components/timezone';
 import PrimaryHeader from './primary-header';
 import Site from 'blocks/site';
+import { localize } from 'i18n-calypso';
+import { updateConciergeInfo } from 'state/concierge/info/actions';
+import { getConciergeInfo } from 'state/selectors';
 
 class InfoStep extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
```

Comment:
Do we actually need to keep this data both in state and in Redux? When the form is updated, could we just dispatch the Redux action instead of updating local state?

Reply:
I didn't want to dispatch an action whenever the customer typed something, I tried throttling but that means the textarea value will also be throttled, am I missing something?

Reply:
This isn't worth holding up this PR, as the product works fine as-is. But so you know, my thinking here is:

1. Whenever I see the same data being duplicated in two places it raises a code smell for me. It causes potential sources of mismatch later on, even if we're careful about it now. For example we may make a change that removes the sync-ing of component state (what the customer sees) with Redux state (what gets sent to the API), causing an invisible bug.

2. It lets us lean on our existing Redux state caching. If the customer refreshes the page mid-typing, it's a nice touch if everything they were writing before comes back to them.

3. When a component accepts a prop, and that prop changes, I usually expect the component to update to reflect that. You're seeding the component with the Redux state when the component mounts, but then ignoring that prop further. Again, right now this works fine, but can be a source of confusion later on.

I wouldn't be worried about the performance issues of dispatching on every test change. [This is exactly what the Happychat client does](https://github.com/Automattic/wp-calypso/blob/master/client/components/happychat/composer.jsx#L43) (the Composer doesn't keep a local copy of the message, it announces changes with `onChange` and then expects the parent component to update state and pass back a new `message` prop).

Let's leave this, as it's working just fine.

Reply:
Ah, you already made the change. :) Cool, thanks.